src/KDSoapClient/KDSoapClientInterface.h

```diff
@@ -346,6 +346,26 @@ class KDSOAP_EXPORT KDSoapClientInterface
       */
     void setTimeout(int msecs);
 
+    /**
+     * @brief setUseWS_Addressing
+     * @param useWS_Addressing
+     * If this parameter is set to true then the outgoing messages SOAP action parameter
+     * will be added as an Action field of the SOAP envelope's header
+     * (instead of appending it to the Content-type HTTP header).
+     * This will make the KDSOAP generated SOAP requests compatible with the WS-Addressing
+     * specification. For a request body example please see:
+     * https://www.w3.org/Submission/ws-addressing/#_Toc77464324
+     */
+    void setUseWS_Addressing(bool useWS_Addressing = true);
+
+    /**
+     * @brief useWS_Addressing
+     * @return
+     * Returns true if the actions of the sent messages are passed in the SOAP envelope's
+     * header section instead of sending them in the request's URL parameters.
+     */
+    bool useWS_Addressing() const;
```

Comment:
Personally I don't like the underscore in the name. I would like useWSAddressingAction.

Reply:
Fixed.